### PR TITLE
API tests

### PR DESCRIPTION
**Pull request by MariusBaldovin**
Added tests for certificate endpoints:
- GET request:  200 (OK)
- POST request: 200 (OK), 400 (Bad Request), 409 (Already Exists)
- DELETE request: 200 (OK), 400 (Bad Request), 404(Not Found)

Added tests for report endpoints:
- GET request:  200 (OK), 404(Not Found)
- POST request: 200 (OK), 404(Not Found)
- DELETE request: 200 (OK), 400 (Bad Request), 404 (Not Found), 500 (Server Error)

Updated api.py:
- added 404 response in get_results if report is missing
-  added 400 response in delete_report if timestamp format is incorrect

### CODE REVIEW

**Comment by jboddey** [comment]
You have a pylint issue to fix

File: testing/api/test_api.py [hunk 1]
```diff
@@ -710,24 +719,99 @@ def test_system_shutdown_in_progress(testrun):  # pylint: disable=W0613
...
+                  "update_available", 
+                  "latest_version", 
+                  "latest_version_url"]
+
+  # Check if all the keys are in json response
+  for key in version_json:
+    assert key in response
+
+  # Check if the update_available is a boolean
+  assert isinstance(response["update_available"], bool)
```
[the remark this answers is not in the document]
Comment (MariusBaldovin) [reply testing/api/test_api.py]: Fixed

File: testing/api/test_api.py [hunk 2]
```diff
@@ -710,24 +719,99 @@ def test_system_shutdown_in_progress(testrun):  # pylint: disable=W0613
...
+    timestamp = timestamp.replace(" ", "T")
+
+    # Create the report folder path
+    report_folder = (main_folder / "reports" / timestamp /
+                      "test" / mac_address)
+
+    # Ensure the report folder exists
+    report_folder.mkdir(parents=True, exist_ok=True)
+
+    # Define the source and target paths for the report.json
```
[the remark this answers is not in the document]
Comment (MariusBaldovin) [reply testing/api/test_api.py]: Fixed, it is copying the folder content

**Comment by jboddey** [comment]
@MariusBaldovin 2 issues. 1) There are some tests in API that are failing 2) We need to figure out why the API tests say that it has passed even though tests have failed.

**Comment by MariusBaldovin** [comment]
@jboddey I will investigate this issue. In my dev branch, all the tests passed after the last commit.

**Review by jboddey — APPROVED** [review]

**Pull request merged**